avm/res/digital-twins/digital-twins-instance/endpoint--event-hub/main.bicep

```diff
@@ -0,0 +1,89 @@
+metadata name = 'Digital Twins Instance EventHub Endpoint'
+metadata description = 'This module deploys a Digital Twins Instance EventHub Endpoint.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Optional. The name of the Digital Twin Endpoint.')
+param name string = 'EventHubEndpoint'
+
+@description('Conditional. The name of the parent Digital Twin Instance resource. Required if the template is used in a standalone deployment.')
+param digitalTwinInstanceName string
+
+@allowed([
+  'IdentityBased'
+  'KeyBased'
+])
+@description('Optional. Specifies the authentication type being used for connecting to the endpoint. If \'KeyBased\' is selected, a connection string must be specified (at least the primary connection string). If \'IdentityBased\' is selected, the endpointUri and entityPath properties must be specified.')
+param authenticationType string = 'IdentityBased'
+
+@description('Optional. Dead letter storage secret for key-based authentication. Will be obfuscated during read.')
+@secure()
+param deadLetterSecret string = ''
+
+@description('Optional. Dead letter storage URL for identity-based authentication.')
+param deadLetterUri string = ''
+
+@description('Conditional. PrimaryConnectionString of the endpoint for key-based authentication. Will be obfuscated during read. Required if the `authenticationType` is "KeyBased".')
+@secure()
+param connectionStringPrimaryKey string = ''
+
+@description('Optional. SecondaryConnectionString of the endpoint for key-based authentication. Will be obfuscated during read. Only used if the `authenticationType` is "KeyBased".')
+@secure()
+param connectionStringSecondaryKey string = ''
+
+@description('Optional. The EventHub name in the EventHub namespace for identity-based authentication.')
+param entityPath string = ''
+
+@description('Optional. The URL of the EventHub namespace for identity-based authentication. It must include the protocol \'sb://\' (i.e. sb://xyz.servicebus.windows.net).')
+param endpointUri string = ''
+
+@description('Optional. The managed identity definition for this resource.')
+param managedIdentities managedIdentitiesType
+
+var identity = !empty(managedIdentities) ? {
```

Comment:
Question: Does this show a warning? If so, I'd suggest to apply the same 'fix' that @krbar implemented in this PR #1238

Reply:
you say implemented but the PR is still open ;)

Reply:
Updated in latets push